doc/developer/design/00000000_template.md

```diff
@@ -3,70 +3,99 @@
+is at the discretion of the author.
+
 Note: Feel free to add or remove sections as needed. However, most design
 docs should at least keep the suggested sections.
 -->
 
-## Context
+## The Problem
 
 <!--
-Bring the reader up to speed enough, such that they can understand the
-following goals and descriptions.
-
-An important reason for this is helping future readers understand the
-assumptions that went into the design, and in turn the goals and design itself.
+What is the user problem we want to solve?
 
-Be sure to capture the customer impact/customer problem, which should be the
-motivation for the proposed design!
+The answer to this question should link to at least one open Github
+issue describing the problem.
 -->
 
-## Goals
+## Success Criteria
 
 <!--
-Enumerate the concrete goals that are in scope for the project.
+What does a solution to this problem need to accomplish in order to
+be successful?
+
+The criteria should help us verify that a proposed solution would solve
+our problem without naming a specific solution. Instead, focus on the
+outcomes we hope results from this work. Feel free to list both qualitative
+and quantitative measurements.
 -->
 
-## Non-Goals
+## Out of Scope
 
 <!--
-Enumerate potential goals that are explicitly _out_ of scope for the project
-ie. what could we do or what do we want to do in the future - but are not doing
-now.
+What does a solution to this problem not need to address in order to be
+successful?
+
+It’s important to be clear about what parts of a problem we won’t be solving
+and why. This leads to crisper designs, and it aids in focusing the reviewer.
 -->
 
-## Overview
+## Solution Proposal
 
 <!--
-Brief, high-level overview. A few sentences long, at most a couple of smaller
-paragraphs.
+What is your preferred solution, and why have you chosen it over the
+alternatives?
+
+This is your opportunity to clearly communicate your chosen design. For any
+design document, the appropriate level of technical details depends both on
+the target reviewers and the nature of the design that is being proposed.
+A good rule of thumb is that you should strive for the minimum level of
+detail that fully communicates the proposal to your reviewers. If you’re
+unsure, reach out to your manager for help.
 -->
 
-## Detailed description
+## Minimal Viable Prototype
 
 <!--
-Describe the approach in detail. If there is no clear frontrunner, feel free to
-list all approaches in alternatives. If applicable, be sure to call out any new
-testing/validation that will be required.
-
-For some features it can be helpful to sketch an implementation. If you're
-working on things that are crossing team boundaries it will be helpful to spell
-out any new interfaces/traits/interactions.
-
-For most new features, you should think about testing, rollout/lifecycle, and
-observability. These things can warrant their own sections.
+Build and share the minimal viable version of your project to validate the
```

Comment:
is this always a prerequisite for creating a new design doc? If not, should it be clarified that this is an optional and/or secondary step after the initial proposal has some eyes on it?

Reply:
I wouldn't want to call it optional, because it is such a valuable step, but I added a note in 8f7dae5cab to leave some room for doing it at a later time (or skipping it) when there is a good reason for it. @JLDLaughlin wdyt?

Reply:
I think we can leave room to skip it if there's a good reason for it. In most cases, though, I'll expect to see something here!

Reply:
Thanks, Jessica. Updated in 63159579ac.